include/boost/hana/experimental/printable.hpp

```diff
@@ -111,6 +111,18 @@ BOOST_HANA_NAMESPACE_BEGIN namespace experimental {
         }
     };
 
+    // model for hana::optional
+    template <>
+    struct print_impl<hana::optional_tag> {
+        template <typename O>
+        static auto apply(O const& optional) {
```

Comment:
Hmm, you're right; `hana::print` documents that it returns `std::string`, yet `print_impl::apply` does not enforce it. I guess the right thing to do is to specify the return type of all the `print_impl::apply`s to be `std::string`. Can you change that while you're at it? Also, I'd use normal return type syntax, since it's more consistent with the rest of the code. I rarely use trailing return type syntax elsewhere in the code (it just happened to be so).